lib/floe/workflow/states/task.rb

```diff
@@ -36,7 +36,7 @@ def start(input)
           super
 
           input          = process_input(input)
-          runner_context = runner.run_async!(resource, input, credentials&.value({}, workflow.credentials))
+          runner_context = runner.run_async!(resource, input, credentials&.value({}, workflow.credentials), context.execution)
```

Comment:
Could we pass the whole context in and then let the runner decide what out of that to pass along?  Thinking that context.current_state could be helpful as well.

Not necessarily here but we should probably enforce that this isn't to be modified by callers (e.g. a built-in method shouldn't be able to set anything in context.execution), and can only set output